app/models/piece.rb

```diff
@@ -91,4 +87,14 @@ def obstructed?(destination_x, destination_y)
     return false unless ((y2 - y1).to_f / (x2 - x1).to_f).abs != 1.0
     'Invalid Input'
   end
+
+  def move_to!(new_x, new_y)
+    if exist?(new_x, new_y)
+      piece_at_destination = game.pieces.find_by(x_position: new_x, y_position: new_y)
+      return 'destination occupied by piece of same color' if color == piece_at_destination.color
+      piece_at_destination.update_attributes(x_position: nil, y_position: nil)
+      true
+    else false
+    end
+  end
```

Comment:
Looking good! 😄 

1 thing missing. Need to update the current pieces position on the board inside this method as well.